tool/loadbalancer/storage.go

```diff
@@ -0,0 +1,55 @@
+package loadbalancer
+
+import (
+	"sort"
+	"sync"
+)
+
+// NewStorage creates a new Storage
```

Comment:
```suggestion
// NewStorage creates a new Storage value.
```
Can you describe what the channel parameter is for? Also mention how the channel is to be services - e.g. what  is the contract b/w sender/receiver - e.g. should the channel be constantly serviced by the storage implementation and should the caller do that (if it's not happening behind the scenes).